plugins/node/instrumentation-react-native-navigation/src/utils/spanCreator.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { AppStateStatus } from 'react-native';
+import { MutableRefObject } from 'react';
+
+import { TracerRef } from './hooks/useTrace';
+import { SpanRef } from './hooks/useSpan';
+import { Attributes, trace, context } from '@opentelemetry/api';
+
+const ATTRIBUTES = {
+  initialView: 'launch',
+  finalView: 'unmount',
+  appState: 'state.end',
+};
+
+const spanStart = (
+  tracer: TracerRef,
+  span: SpanRef,
+  currentRouteName: string,
+  customAttributes?: Attributes,
+  isLaunch?: boolean
+) => {
+  if (!tracer.current || span.current !== null) {
+    // do nothing in case for some reason the tracer is not initialized or there is already an active span
+    return;
+  }
+
+  // Starting the span
+  span.current = tracer.current.startSpan(currentRouteName);
+
+  trace.setSpan(context.active(), span.current);
```

Comment:
this wouldn't update the current active context but rather return a new context with this span stored in it: https://github.com/open-telemetry/opentelemetry-js/blob/main/api/src/trace/context-utils.ts#L51

I don't think there'd really be a great way to set this span as the active parent for any other spans that got created in the view, the only thing exposed by the JS API would require wrapping everything that happens within the view under a callback: https://github.com/open-telemetry/opentelemetry-js/blob/main/api/src/trace/tracer.ts#L85

Joaquin hit something similar with react-router instrumentation could discuss with him I believe he just added a hook to grab the current view span from the instrumentation so that it could be used as a parent span if the user wanted, though for now I think we could just remove this line and think about that functionality later